### ISSUE TITLE
AppleClang detection on macOS

### ISSUE THREAD

**Issue by jdumas** [embree/embree#159]
Since CMake 3.0, the `CMAKE_CXX_COMPILER_ID` for Apple's Clang has changed from `Clang` to `AppleClang`. This mean detecting the Clang compiler should be take this into account. See this [SO thread](https://stackoverflow.com/questions/10046114/in-cmake-how-can-i-test-if-the-compiler-is-clang) and the corresponding [CMake policy (CMP0025)](https://cmake.org/cmake/help/v3.0/policy/CMP0025.html#policy:CMP0025).

For Embree, this means that [this line](https://github.com/embree/embree/blob/b4e6649387e841d48a07e3e765eb2af73a263598/CMakeLists.txt#L268) should be changed to the following:

```cmake
  ELSEIF (CMAKE_CXX_COMPILER_ID MATCHES "Clang")
```

Please note that the current CMake build works because it has the `CMAKE_MINIMUM_REQUIRED` version set to 2.8.11. However, if you were to include embree from a parent project with a minimum version set to 3.0, the CMP0025 policy would be set to `NEW` and Embree's CMake would throw an error.

**Comment by svenwoop** [comment]
Fixed in devel branch.

**Comment by ikoruk** [comment]
@svenwoop This is actually not fixed in the latest version. When `CMAKE_CXX_COMPILER_ID` is `AppleClang` then the cmake build returns:
```
CMake Error at /usr/local/source/libigl/external/embree/CMakeLists.txt:207 (MESSAGE):
  Unsupported compiler: AppleClang
```

**Comment by svenwoop** [comment]
I see, some changes to the compiler handling reverted that fix. I cannot reproduce this issue for some reason, but the following alpha version should fix this: v3.2.2-alpha.1
